src/System.Runtime/tests/System/Enum.cs

```diff
@@ -278,636 +222,262 @@ public static void TestCompareTo()
+    }
 
-        s = ((SimpleEnum)99).ToString("G");
-        Assert.Equal(s, "99");
-
-        s = ((SimpleEnum)0).ToString("G");
-        Assert.Equal(s, "0");  // Not found.
-
-        s = ((ByteEnum)0).ToString("G");
-        Assert.Equal(s, "Min");  // Found
-
-        s = ((ByteEnum)3).ToString("G");
-        Assert.Equal(s, "3");  // No [Flags] attribute
-
-        // Larger values take precedence (and remove the bits from consideration.)
-        s = ((ByteEnum)0xff).ToString("G");
-        Assert.Equal(s, "Max");  // Found
-
-        // An enum with [Flags]
-        s = (AttributeTargets.Class | AttributeTargets.Delegate).ToString("G");
-        Assert.Equal(s, "Class, Delegate");
+    [Theory]
+    [InlineData(SimpleEnum.Red, "Red")]
+    [InlineData(1, "Red")] // Underlying integral
+    public static void TestFormat(object value, string expected)
+    {
+        String s = Enum.Format(typeof(SimpleEnum), value, "F");
+        Assert.Equal(expected, s);
     }
 
     [Fact]
-    public static void TestFormat()
+    public static void TestFormatInvalid()
     {
-        String s;
-
-        s = Enum.Format(typeof(SimpleEnum), SimpleEnum.Red, "F");
-        Assert.Equal(s, "Red");
-
-        // Can pass enum or exact underlying integral.
-        s = Enum.Format(typeof(SimpleEnum), 1, "F");
-        Assert.Equal(s, "Red");
-
-        Assert.Throws<ArgumentNullException>(() => Enum.Format(null, (Int32Enum)1, "F"));
-        Assert.Throws<ArgumentNullException>(() => Enum.Format(typeof(SimpleEnum), null, "F"));
-
-        // Not an enum type.
-        Assert.Throws<ArgumentException>(() => Enum.Format(typeof(Object), 1, "F"));
-
-        // Wrong enumType.
-        Assert.Throws<ArgumentException>(() => Enum.Format(typeof(SimpleEnum), (Int32Enum)1, "F"));
-
-        // Wrong integral.
-        Assert.Throws<ArgumentException>(() => Enum.Format(typeof(SimpleEnum), (short)1, "F"));
-
-        // Not an integral.
-        Assert.Throws<ArgumentException>(() => Enum.Format(typeof(SimpleEnum), "Red", "F"));
+        Assert.Throws<ArgumentNullException>("enumType", () => Enum.Format(null, (Int32Enum)1, "F")); // Enum type is null
+        Assert.Throws<ArgumentNullException>("value", () => Enum.Format(typeof(SimpleEnum), null, "F")); // Value is null
+        
+        Assert.Throws<ArgumentException>("enumType", () => Enum.Format(typeof(Object), 1, "F")); // Enum type is not an enum type
+        
+        Assert.Throws<ArgumentException>(null, () => Enum.Format(typeof(SimpleEnum), (Int32Enum)1, "F")); // Value is of the wrong enum type
+        
+        Assert.Throws<ArgumentException>(null, () => Enum.Format(typeof(SimpleEnum), (short)1, "F")); // Value is of the wrong integral
+        Assert.Throws<ArgumentException>(null, () => Enum.Format(typeof(SimpleEnum), "Red", "F")); // Value is of the wrong integral
     }
 
-    private enum SimpleEnum
+    public enum SimpleEnum
```

Comment:
Yup, since it is now InlineData, SimpleEnum is a parameter of a public function so has to be public itself